vortexor/src/vortexor.rs

```diff
@@ -0,0 +1,216 @@
+            num_quic_endpoints.try_into().unwrap(),
+            quic_config,
+        )
+        .unwrap();
+
+        TpuSockets {
+            tpu_quic,
+            tpu_quic_fwd,
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_vortexor(
+        tpu_sockets: TpuSockets,
+        staked_nodes: Arc<RwLock<StakedNodes>>,
+        tpu_sender: Sender<PacketBatch>,
+        tpu_fwd_sender: Sender<PacketBatch>,
+        max_connections_per_peer: u64,
+        max_tpu_staked_connections: u64,
+        max_tpu_unstaked_connections: u64,
+        max_streams_per_ms: u64,
+        max_connections_per_ipaddr_per_min: u64,
+        tpu_coalesce: Duration,
+        identity_keypair: &Keypair,
+        exit: Arc<AtomicBool>,
+    ) -> Self {
+        let config = TpuStreamerConfig {
+            tpu_thread_name: "solQuicTpu",
+            tpu_metrics_name: "quic_streamer_tpu",
+            tpu_fwd_thread_name: "solQuicTpuFwd",
+            tpu_fwd_metrics_name: "quic_streamer_tpu_forwards",
+            quic_server_params: QuicServerParams {
+                max_connections_per_peer: max_connections_per_peer.try_into().unwrap(),
+                max_staked_connections: max_tpu_staked_connections.try_into().unwrap(),
+                max_unstaked_connections: max_tpu_unstaked_connections.try_into().unwrap(),
+                max_streams_per_ms,
+                max_connections_per_ipaddr_per_min,
+                wait_for_chunk_timeout: DEFAULT_WAIT_FOR_CHUNK_TIMEOUT,
+                coalesce: tpu_coalesce,
+            },
+        };
+
+        Vortexor::new(
+            identity_keypair,
+            tpu_sockets,
+            tpu_sender,
+            tpu_fwd_sender,
+            staked_nodes,
+            config,
+            exit,
+        )
+    }
+
+    /// Create a new TPU Vortexor
+    pub fn new(
+        keypair: &Keypair,
+        tpu_sockets: TpuSockets,
+        tpu_sender: Sender<PacketBatch>,
+        tpu_fwd_sender: Sender<PacketBatch>,
+        staked_nodes: Arc<RwLock<StakedNodes>>,
+        config: TpuStreamerConfig,
+        exit: Arc<AtomicBool>,
+    ) -> Self {
+        let TpuSockets {
+            tpu_quic,
+            tpu_quic_fwd,
+        } = tpu_sockets;
+
+        let TpuStreamerConfig {
+            tpu_thread_name,
+            tpu_metrics_name,
+            tpu_fwd_thread_name,
+            tpu_fwd_metrics_name,
+            mut quic_server_params,
+        } = config;
+
+        let tpu_result = spawn_server_multi(
+            tpu_thread_name,
+            tpu_metrics_name,
+            tpu_quic,
+            keypair,
+            tpu_sender.clone(),
+            exit.clone(),
+            staked_nodes.clone(),
+            quic_server_params.clone(),
+        )
+        .unwrap();
+
+        // Fot TPU forward -- we disallow unstaked connections. Allocate all connection resources
+        // for staked connections:
+        quic_server_params.max_staked_connections = quic_server_params
```

Comment:
It's confusing as a user of vortexor that we are summing up staked and unstaked count on the forwarding streamer.
Should we add another CLI arg specific for the max connections on forwarding streamer?

Reply:
Will do